src/main/java/net/sf/javaanpr/gui/windows/FrameMain.java

```diff
@@ -277,20 +277,25 @@ public RecognizeThread(FrameMain parentFrame) {
             this.parentFrame = parentFrame;
         }
 
+        private void setFailedAndPrintStackTrace(Exception exception) {
+            this.parentFrame.recognitionLabel.setText("failed");
+            exception.printStackTrace();
+        }
+
         @Override
         public void run() {
             String recognizedText = "";
             this.parentFrame.recognitionLabel.setText("processing...");
             int index = this.parentFrame.selectedIndex;
             try {
                 recognizedText = Main.systemLogic.recognize(this.parentFrame.car, false);
-            } catch (Exception ex) { // TODO exception
-                this.parentFrame.recognitionLabel.setText("failed");
-                ex.printStackTrace();
-                return;
+                this.parentFrame.recognitionLabel.setText(recognizedText);
```

Comment:
Is there any reason why you moved these two UI calls into the try block? Other than the fact that you can then omit two `return` statements in the catch blocks. If not, I'd be more comfortable adding the two `return`s and moving this back out of the try block. It shouldn't really matter execution-wise, but it's cleaner to have as short try blocks as possible.